woocommerce/woocommerce-gateway-stripe
# Replace jQuery shorthands with their full counterparts

Make sure #1392 is merged before this PR to bring the composer v2 updates to `master`.

# Changes proposed in this Pull Request:

Fixes #1390

This removes all calls to the jQuery shorthands we've used in our code. I located the shorthands in use with the following regex, as described in [this comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1390#issuecomment-738917808):

```regex
\.(blur|focus|focusin|focusout|resize|scroll|click|dblclick|mousedown|mouseup|mousemove|mouseover|mouseout|mouseenter|mouseleave|change|select|submit|keydown|keypress|keyup|contextmenu)\(
```


# Testing instructions

- Make sure the `.on( 'change' )` handlers are working as intended for the Payment Request buttons. I'm honestly not sure how to properly test this, but I _think_ it's just making sure that they render with the correct brand?
- Make sure that the credit card input loses focus when an error occurs during checkout (the `blur` action).
- I'm not sure what the `submit` trigger is doing in `sourceResponse`, but I assume there's some way to test this?

Here's the `sourceResponse` function, for a quick glance:

```js
		sourceResponse: function( response ) {
			if ( response.error ) {
				return $( document.body ).trigger( 'stripeError', response );
			}
			wc_stripe_form.reset();
			wc_stripe_form.form.append(
				$( '<input type="hidden" />' )
					.addClass( 'stripe-source' )
					.attr( 'name', 'stripe_source' )
					.val( response.source.id )
			);
			if ( $( 'form#add_payment_method' ).length ) {
				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
			}

			wc_stripe_form.form.submit();
			wc_stripe_form.form.trigger( 'submit' );
		},
```

-------------------
